pennylane/debugging.py

```diff
@@ -87,6 +99,52 @@ def circuit():
     2: array([0.70710678+0.j, 0.        +0.j, 0.        +0.j, 0.70710678+0.j]),
     'execution_results': 0.0}
     """
+    new_tapes = []
+    accumulated_ops = []
+    snapshot_tags = []
+
+    for op in tape.operations:
+        if isinstance(op, qml.Snapshot):
+            snapshot_tags.append(op.tag or len(new_tapes))
+
+            if op.hyperparameters["use_device_shots"]:
+                shots = tape.shots
+            elif op.hyperparameters["shots"]:
+                shots = qml.measurements.Shots(op.hyperparameters["shots"])
+            else:
+                # Assume StateMeasurement
+                shots = None
+
+            meas_op = op.hyperparameters["measurement"]
+            tape_meas = [meas_op] if meas_op else [qml.state()]
+            new_tapes.append(QuantumTape(ops=accumulated_ops, measurements=tape_meas, shots=shots))
```

Comment:
```suggestion
            new_tapes.append(type(tape)(ops=accumulated_ops, measurements=tape_meas, shots=shots))
```
Will preserve the tape/ qscript distinction so we don't have to worry about our workflow becoming multiprocessing-incompatible due to `QuantumTape`'s threading lock.

Hopefully we can deprecate the queuing behaviour/threading lock from `QuantumTape` in the next release cycle and not have to worry about this anymore, but for now, we should be cautious about introducing `QuantumTape` when we don't need to.